apis/v1/gateway_types.go

```diff
@@ -701,10 +698,16 @@ const (
 	// true.
 	GatewayReasonProgrammed GatewayConditionReason = "Programmed"
 
-	// This reason is used with the "Programmed" and "Accepted" conditions when the Gateway is
-	// syntactically or semantically invalid.
+	// This reason is used with the "Programmed" and "Accepted" conditions when
+	// the Gateway is syntactically or semantically invalid.
 	GatewayReasonInvalid GatewayConditionReason = "Invalid"
 
+	// This reason is used with the "Programmed" condition when the Gateway is
+	// not programmed because the TLS config is invalid. This could include
+	// unspecified TLS configuration, or some unrecognized or invalid values in
+	// the TLS configuration.
+	GatewayReasonInvalidTLSConfig GatewayConditionReason = "InvalidTLSConfig"
```

Comment:
This feels a bit too similar to `Invalid` to warrant its own `Reason`. `InvalidTLSConfig` feels like a string details extension of `Invalid`, rather than its own `Reason`. This is definitely more of a nit that anything, but creating a new Reason which is a superset of another `Reason` could lead to `Reason` bloat

Reply:
Good call, decided to drop this entirely.